### PR TITLE
Latest updates

### PR DESCRIPTION
**Pull request by joerivandervelde**
Added fixes, tests, helper tools, plotting functions

### CODE REVIEW

File: src/main/java/org/molgenis/data/annotation/reportrvcf/MultiPhenotypeCohortAnalysis.java [hunk 1]
```diff
@@ -109,10 +107,10 @@ public void countAffectedPerGene(HashMap<String, String> geneToChromPos, HashMap
...
                 for(String sample : rvcf.getSampleStatus().keySet()) {
 
-                    if(sampleAddedForGene.contains(sample+"_"+gene))
-                    {
-                        continue;
-                    }
+//                    if(sampleAddedForGene.contains(sample+"_"+gene))
+//                    {
+//                        continue;
+//                    }
```
Comment (bartcharbon) [inline src/main/java/org/molgenis/data/annotation/reportrvcf/MultiPhenotypeCohortAnalysis.java]: remove? or uncomment?

File: src/main/java/org/molgenis/data/annotation/reportrvcf/MultiPhenotypeCohortAnalysis.java [hunk 2]
```diff
@@ -124,7 +122,7 @@ public void countAffectedPerGene(HashMap<String, String> geneToChromPos, HashMap
                         geneAndPhenotypeToAffectedCount.put(gene, phenotype, count);
 
                         // make sure we count an individual only once per gene
-                        sampleAddedForGene.add(sample+"_"+gene);
+             //           sampleAddedForGene.add(sample+"_"+gene);
```
Comment (bartcharbon) [inline src/main/java/org/molgenis/data/annotation/reportrvcf/MultiPhenotypeCohortAnalysis.java]: remove? or uncomment?

**Comment by joerivandervelde** [comment]
Please review again!

File: src/main/java/org/molgenis/data/annotation/reportrvcf/HomHetCounts.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    private VcfRepository vcf;
+    private PrintWriter pw;
+    int nrOfSamples;
+
+
+    public static String HEADER = "Gene" + "\t" + "HetAbs" + "\t" + "HomAbs" + "\t" + "HomFrac" + "\t" + "HetFrac";
+
+    public static void main(String[] args) throws Exception {
+        HomHetCounts hhc = new HomHetCounts(new File("/Users/joeri/Desktop/GAVIN-APP/1000G_diag_FDR/exomePlus/ALL.chr1to22plusXYMT_RVCF_r1.0.vcf"),
+                new File("/Users/joeri/Desktop/GAVIN-APP/1000G_diag_FDR/exomePlus/HomHetCounts_r1.0.tsv"),
```
Comment (bartcharbon) [inline src/main/java/org/molgenis/data/annotation/reportrvcf/HomHetCounts.java]: hardcoded path, path should be an argument

File: src/main/R/FDR_plot.R [hunk 4]
```diff
@@ -1,26 +1,39 @@
 library(ggplot2)
 
 # load FDR and CGD sets
-fdr <- read.table("/Users/joeri/github/rvcf/src/test/resources/bundle_r1.0/FDR_allGenes.tsv", sep="\t", header=T)
-cgd <- read.table("/Users/joeri/github/rvcf/src/test/resources/GenesInheritance.tsv", sep="\t", header=T)
+fdr <- read.table("/Users/joeri/github/rvcf/src/test/resources/bundle_r1.0/FDR_allGenes_r1.0.tsv", sep="\t", header=T)
+cgd <- read.table("/Users/joeri/github/rvcf/src/test/resources/GenesInheritance31aug2016.tsv", sep="\t", header=T)
+
```
Comment (bartcharbon) [inline src/main/R/FDR_plot.R]: hardcoded paths, not sure how to fix in R, can those be relative to the project?

File: src/main/R/FOR_plot.R [hunk 5]
```diff
@@ -0,0 +1,19 @@
+library(ggplot2)
+
+# load FOR and CGD sets
+falseOmRate <- read.table("/Users/joeri/Desktop/GAVIN-APP/1000G_diag_FDR/FOR/FOR_results_per_gene_r1.0.tsv", sep="\t", header=T)
+cgd <- read.table("/Users/joeri/github/rvcf/src/test/resources/GenesInheritance31aug2016.tsv", sep="\t", header=T)
+
```
Comment (bartcharbon) [inline src/main/R/FOR_plot.R]: hardcoded paths, not sure how to fix in R, can those be relative to the project?

File: src/main/R/HHC_plot.R [hunk 6]
```diff
@@ -0,0 +1,55 @@
+library(ggplot2)
+
+# load HHC and CGD sets
+hhc <- read.table("/Users/joeri/github/rvcf/src/test/resources/HomHetCounts_r1.0.tsv", sep="\t", header=T)
+cgd <- read.table("/Users/joeri/github/rvcf/src/test/resources/GenesInheritance31aug2016.tsv", sep="\t", header=T)
+
```
Comment (bartcharbon) [inline src/main/R/HHC_plot.R]: hardcoded paths, not sure how to fix in R, can those be relative to the project?

File: src/main/java/org/molgenis/data/annotation/reportrvcf/Reactome.java [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+            if(split.length < 4)
+            {
+                throw new Exception("Bad data, less than 4 elements: " + pathwayName);
+            }
+            Set<String> genes = new HashSet<>();
+            for(int i = 3; i < split.length; i++)
+            {
+                genes.add(split[i]);
+            }
+            //System.out.println("Pathway " + pathwayName + " has genes " + genes.toString());
```
Comment (bartcharbon) [inline src/main/java/org/molgenis/data/annotation/reportrvcf/Reactome.java]: outcommented log line can be removed?

File: src/main/java/org/molgenis/data/annotation/reportrvcf/ReactomeMultiPhenotypeCohortAnalysis.java [hunk 8]
```diff
@@ -0,0 +1,270 @@
...
+        mdca.start();
+    }
+
+    public ReactomeMultiPhenotypeCohortAnalysis(File rvcfInputFile, File outputZscoreFile, File reactomeFile) throws Exception {
+        this.rvcfInputFile = rvcfInputFile;
+        this.outputZscoreFile = outputZscoreFile;
+
+        /** Reactome mapping **/
+//        this.pathwayToGenes = Reactome.load(reactomeFile);
+//        this.geneToPathways = Reactome.makeGeneToPathwaysMap(pathwayToGenes);
```
Comment (bartcharbon) [inline src/main/java/org/molgenis/data/annotation/reportrvcf/ReactomeMultiPhenotypeCohortAnalysis.java]: Replace outcommented code with "TODO"

File: src/main/java/org/molgenis/data/annotation/reportrvcf/ReactomeMultiPhenotypeCohortAnalysis.java [hunk 9]
```diff
@@ -0,0 +1,270 @@
...
+
+        /** OmimHpo mapping **/
+        this.pathwayToGenes = OmimHpo.load(reactomeFile);
+        filter();
+        this.geneToPathways = OmimHpo.makeGeneToHpo(pathwayToGenes);
+
+        /** CGD mapping **/
+//        Map<String, CGDEntry> cgd = LoadCGD.loadCGD(reactomeFile);
+//        this.pathwayToGenes = LoadCGD.makeCgdToGenesMap(cgd);
+//        this.geneToPathways = LoadCGD.makeGenesToDiseaseName(cgd);
```
Comment (bartcharbon) [inline src/main/java/org/molgenis/data/annotation/reportrvcf/ReactomeMultiPhenotypeCohortAnalysis.java]: Replace outcommented code with "TODO"

File: src/main/java/org/molgenis/data/annotation/reportrvcf/ReactomeMultiPhenotypeCohortAnalysis.java [hunk 10]
```diff
@@ -0,0 +1,270 @@
...
+        // transform fractions into Z-scores by comparing 1 phenotype group to the others
+        MultiKeyMap pathwayAndPhenotypeToAffectedZscores = new MultiKeyMap();
+
+        // execution
+        phenotypeToNrOfIndividualsFromInputVcf(individualsToPhenotype, phenotypeToNrOfIndividuals, this.rvcfInputFile);
+        System.out.println("individualsToPhenotype: " + individualsToPhenotype.toString());
+        System.out.println("phenotypeToNrOfIndividuals: " + phenotypeToNrOfIndividuals.toString());
+
+        countAffectedPerPathway(individualsToPhenotype, pathwayAndPhenotypeToAffectedCount, this.rvcfInputFile, this.geneToPathways);
+  //      System.out.println("countAffectedPerGene: " + pathwayAndPhenotypeToAffectedCount.toString());
```
Comment (bartcharbon) [inline src/main/java/org/molgenis/data/annotation/reportrvcf/ReactomeMultiPhenotypeCohortAnalysis.java]: remove outcommented log line

File: src/main/java/org/molgenis/data/annotation/reportrvcf/ReactomeMultiPhenotypeCohortAnalysis.java [hunk 11]
```diff
@@ -0,0 +1,270 @@
...
+
+//
```
Comment (bartcharbon) [inline src/main/java/org/molgenis/data/annotation/reportrvcf/ReactomeMultiPhenotypeCohortAnalysis.java]: remove empty comment

File: src/main/java/org/molgenis/data/annotation/reportrvcf/ReactomeMultiPhenotypeCohortAnalysis.java [hunk 12]
```diff
@@ -0,0 +1,270 @@
...
+        convertToZscore(pathwayAndPhenotypeToAffectedCount, pathwayAndPhenotypeToAffectedZscores, phenotypeToNrOfIndividuals.keySet());
+//
```
Comment (bartcharbon) [inline src/main/java/org/molgenis/data/annotation/reportrvcf/ReactomeMultiPhenotypeCohortAnalysis.java]: remove empty comment

File: src/main/java/org/molgenis/data/annotation/reportrvcf/ReactomeMultiPhenotypeCohortAnalysis.java [hunk 13]
```diff
@@ -0,0 +1,270 @@
...
+        }
+    }
+
+
+    public void countAffectedPerPathway(HashMap<String, String> individualsToPhenotype, MultiKeyMap pathwayAndPhenotypeToAffectedCount, File rvcfInputFile, HashMap<String, Set<String>> geneToPathways) throws Exception {
+        VcfRepository vcf = new VcfRepository(rvcfInputFile, "vcf");
+        Iterator<Entity> vcfIterator = vcf.iterator();
+
+        // "SampleID_GeneName"
+ //       Set<String> sampleAddedForGene = new HashSet<>();
```
Comment (bartcharbon) [inline src/main/java/org/molgenis/data/annotation/reportrvcf/ReactomeMultiPhenotypeCohortAnalysis.java]: remove outcommented code

File: src/main/java/org/molgenis/data/annotation/reportrvcf/ReactomeMultiPhenotypeCohortAnalysis.java [hunk 14]
```diff
@@ -0,0 +1,270 @@
...
+            {
+                String gene = rvcf.getGene();
+
+                if(!geneToPathways.containsKey(gene))
+                {
+                    continue;
+                }
+                Set<String> pathways = geneToPathways.get(gene);
+
+            //    System.out.println("looking at gene: " + gene + " having pathways " + pathways.toString());
```
Comment (bartcharbon) [inline src/main/java/org/molgenis/data/annotation/reportrvcf/ReactomeMultiPhenotypeCohortAnalysis.java]: remove outcommented log line

File: src/main/java/org/molgenis/data/annotation/reportrvcf/ReactomeMultiPhenotypeCohortAnalysis.java [hunk 15]
```diff
@@ -0,0 +1,270 @@
...
+
+                for(String sample : rvcf.getSampleStatus().keySet()) {
+//
+                    // overrepresentation vs amount presumed affected?
+//                    if(sampleAddedForGene.contains(sample+"_"+gene))
+//                    {
+//                        continue;
+//                    }
```
Comment (bartcharbon) [inline src/main/java/org/molgenis/data/annotation/reportrvcf/ReactomeMultiPhenotypeCohortAnalysis.java]: remove outcommented code, replace with "TODO"

File: src/main/java/org/molgenis/data/annotation/reportrvcf/ReactomeMultiPhenotypeCohortAnalysis.java [hunk 16]
```diff
@@ -0,0 +1,270 @@
...
+                if(!geneToPathways.containsKey(gene))
+                {
+                    continue;
+                }
+                Set<String> pathways = geneToPathways.get(gene);
+
+            //    System.out.println("looking at gene: " + gene + " having pathways " + pathways.toString());
+
+                for(String sample : rvcf.getSampleStatus().keySet()) {
+//
```
Comment (bartcharbon) [inline src/main/java/org/molgenis/data/annotation/reportrvcf/ReactomeMultiPhenotypeCohortAnalysis.java]: remove empty log line

File: src/main/java/org/molgenis/data/annotation/reportrvcf/ReactomeMultiPhenotypeCohortAnalysis.java [hunk 17]
```diff
@@ -0,0 +1,270 @@
...
+                    // overrepresentation vs amount presumed affected?
+//                    if(sampleAddedForGene.contains(sample+"_"+gene))
+//                    {
+//                        continue;
+//                    }
+
+                    if (MatchVariantsToGenotypeAndInheritance.status.isPresumedAffected(rvcf.getSampleStatus().get(sample)))
+                    //if (rvcf.getSampleStatus().get(sample).toString().contains("AFFECTED"))
```
Comment (bartcharbon) [inline src/main/java/org/molgenis/data/annotation/reportrvcf/ReactomeMultiPhenotypeCohortAnalysis.java]: remove

File: src/main/java/org/molgenis/data/annotation/reportrvcf/ReactomeMultiPhenotypeCohortAnalysis.java [hunk 18]
```diff
@@ -0,0 +1,270 @@
...
+
+                        for(String pathway : pathways)
+                        {
+                            Integer count = pathwayAndPhenotypeToAffectedCount.containsKey(pathway, phenotype) ? (Integer)pathwayAndPhenotypeToAffectedCount.get(pathway, phenotype) : 0;
+                            count++;
+                            pathwayAndPhenotypeToAffectedCount.put(pathway, phenotype, count);
+                        }
+
+                        // make sure we count an individual only once per gene
+    //                    sampleAddedForGene.add(sample+"_"+gene);
```
Comment (bartcharbon) [inline src/main/java/org/molgenis/data/annotation/reportrvcf/ReactomeMultiPhenotypeCohortAnalysis.java]: remove, replace with TODO

**Pull request merged**